rowenc: adjust TestInitIndexFetchSpec to work with test tenant

it currently fails with
```
--- FAIL: TestInitIndexFetchSpec (0.91s)
    test_log_scope.go:167: test logs captured to: /Users/yuzefovich/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/05aa764b9d9d6f0981d84c80dece8bc5/logTestInitIndexFetchSpec1528729971
    test_log_scope.go:81: use -show-logs to present logs inline
    datadriven.go:144: 
        /private/var/tmp/_bazel_yahor/e311b1c2454b7c301a6b73bddff450e7/sandbox/darwin-sandbox/3876/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/rowenc/rowenc_test_/rowenc_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/rowenc/testdata/index-fetch:16:
         table: t
        index: t_pkey
        columns:
          - a
          - b
          - c
        output didn't match expected:
        @@ -8,11 +8,11 @@
           "is_unique_index": true,
           "geo_config": {},
           "encoding_type": 1,
           "num_key_suffix_columns": 0,
           "max_keys_per_row": 1,
        -  "key_prefix_length": 2,
        +  "key_prefix_length": 4,
           "max_family_id": 0,
           "family_default_columns": null,
           "key_and_suffix_columns": [
             {
               "column": {
    panic.go:522: -- test log scope end --
FAIL
```
We probably need to create a separate testdata file.

--- a/pkg/sql/flowinfra/cluster_test.go
+++ b/pkg/sql/flowinfra/cluster_test.go
@@ -506,6 +506,10 @@ func TestLimitedBufferingDeadlock(t *testing.T) {
 // Test that DistSQL reads fill the BatchRequest.Header.GatewayNodeID field with
 // the ID of the gateway (as opposed to the ID of the node that created the
 // batch). Important to lease follow-the-workload transfers.
+//
+// Note that in single tenant and shared-process modes, GatewayNodeID field is
+// set on the kvclient side (i.e. in the SQL pod) whereas in external-process
+// mode it's set on the kvserver side (i.e. in the storage pod).
 func TestDistSQLReadsFillGatewayID(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
@@ -522,8 +526,7 @@ func TestDistSQLReadsFillGatewayID(t *testing.T) {
 		base.TestClusterArgs{
 			ReplicationMode: base.ReplicationManual,
 			ServerArgs: base.TestServerArgs{
-				UseDatabase:       "test",
-				DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(109392),
+				UseDatabase: "test",
 				Knobs: base.TestingKnobs{Store: &kvserver.StoreTestingKnobs{
 					EvalKnobs: kvserverbase.BatchEvalTestingKnobs{
 						TestingEvalFilter: func(filterArgs kvserverbase.FilterArgs) *kvpb.Error {
@@ -531,10 +534,10 @@ func TestDistSQLReadsFillGatewayID(t *testing.T) {
 							if !ok {
 								return nil
 							}
-							if !strings.HasPrefix(
-								scanReq.Key.String(),
-								fmt.Sprintf("/Table/%d/1", tableID.Load()),
-							) {
+							pkPrefix := fmt.Sprintf("/Table/%d/1", tableID.Load())
+							tenantPKPrefix := fmt.Sprintf("/Tenant/%d%s", serverutils.TestTenantID().InternalValue, pkPrefix)
+							if !strings.HasPrefix(scanReq.Key.String(), pkPrefix) &&
+								!strings.HasPrefix(scanReq.Key.String(), tenantPKPrefix) {
 								return nil
 							}
 
@@ -551,6 +554,13 @@ func TestDistSQLReadsFillGatewayID(t *testing.T) {
 			},
 		})
 	defer tc.Stopper().Stop(context.Background())
+
+	if tc.DefaultTenantDeploymentMode().IsExternal() {
+		tc.GrantTenantCapabilities(
+			context.Background(), t, serverutils.TestTenantID(),
+			map[tenantcapabilitiespb.ID]string{tenantcapabilitiespb.CanAdminRelocateRange: "true"},
+		)
+	}
 
 	db := tc.ServerConn(0)
 	sqlutils.CreateTable(t, db, "t",

--- a/pkg/sql/rowenc/index_encoding_test.go
+++ b/pkg/sql/rowenc/index_encoding_test.go
@@ -40,6 +40,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/json"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/randutil"
 	"github.com/cockroachdb/cockroach/pkg/util/trigram"
 	"github.com/cockroachdb/cockroach/pkg/util/vector"
@@ -1076,7 +1077,6 @@ func TestEncodeTrigramInvertedIndexSpans(t *testing.T) {
 
 	runTest := func(indexedValue, value string, searchType trigramSearchType,
 		expectContainsKeys, expected, expectUnique bool) {
-		t.Logf("test case: %s %s %v %t %t %t", indexedValue, value, searchType, expectContainsKeys, expected, expectUnique)
 		keys, err := EncodeInvertedIndexTableKeys(tree.NewDString(indexedValue), nil, descpb.LatestIndexDescriptorVersion)
 		require.NoError(t, err)
 
@@ -1285,6 +1285,7 @@ func TestDecodeKeyVals(t *testing.T) {
 // write leaf keys, so that's what's tested here.
 func TestVectorEncoding(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
 	srv, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
@@ -1376,6 +1377,7 @@ func TestVectorEncoding(t *testing.T) {
 
 func TestVectorCompositeEncoding(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
 	srv, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})

--- a/pkg/sql/rowenc/index_fetch_test.go
+++ b/pkg/sql/rowenc/index_fetch_test.go
@@ -28,9 +28,7 @@ func TestInitIndexFetchSpec(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	srv, db, kvDB := serverutils.StartServer(t, base.TestServerArgs{
-		DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(109390),
-	})
+	srv, db, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
 	defer srv.Stopper().Stop(context.Background())
 	codec := srv.ApplicationLayer().Codec()
 
@@ -75,6 +73,13 @@ func TestInitIndexFetchSpec(t *testing.T) {
 				var spec fetchpb.IndexFetchSpec
 				if err := rowenc.InitIndexFetchSpec(&spec, codec, table, index, fetchColumnIDs); err != nil {
 					d.Fatalf(t, "%+v", err)
+				}
+				if srv.StartedDefaultTestTenant() {
+					// If we started a test tenant, then all keys will have the
+					// tenant prefix (of 2 bytes) prepended to them. In order to
+					// not create the duplicate expectation file, we'll adjust
+					// the spec here to ignore those 2 bytes.
+					spec.KeyPrefixLength -= uint32(len(codec.TenantPrefix()))
 				}
 				res, err := json.MarshalIndent(&spec, "", "  ")
 				if err != nil {

